workflows/bamtofastq.nf

```diff
@@ -0,0 +1,316 @@
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    VALIDATE INPUTS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+def summary_params = NfcoreSchema.paramsSummaryMap(workflow, params)
+
+// Validate input parameters
+WorkflowBamtofastq.initialise(params, log)
+
+// Check input path parameters to see if they exist
+def checkPathParamList = [
+    params.fasta,
+    params.fasta_fai,
+    params.input,
+    params.multiqc_config
+    ]
+
+for (param in checkPathParamList) { if (param) { file(param, checkIfExists: true) } }
+
+// Check mandatory parameters
+if (params.input) { ch_input = extract_csv(file(params.input, checkIfExists: true)) } else { exit 1, 'Input samplesheet not specified!' }
+
+
+// Initialize file channels based on params
+fasta       = params.fasta      ? Channel.fromPath(params.fasta).collect()      : Channel.value([])
+fasta_fai   = params.fasta_fai  ? Channel.fromPath(params.fasta_fai).collect()  : Channel.value([])
+
+// Initialize value channels based on params
+chr         = params.chr        ?: Channel.empty()
```

Comment:
```suggestion
fasta     = params.fasta     ? Channel.fromPath(params.fasta).collect()      : Channel.value([])
fasta_fai = params.fasta_fai ? Channel.fromPath(params.fasta_fai).collect()  : Channel.value([])

// Initialize value channels based on params
chr       = params.chr       ? : Channel.empty()
```

Reply:
If one inserts a whitespace after the `? :` the pipeline will fail with this Error:
```
- cause: Unexpected input: '?' @ line 31, column 30.
   chr       = params.chr       ? : Channel.empty()
```